tiledb/common/logger.cc

```diff
@@ -345,4 +349,12 @@ void LOG_ERROR(const std::stringstream& msg) {
   global_logger().error(msg);
 }
 
+/**
+ * Returns whether trace-level or above is in effect.  Useful for callsites which want to avoid
+ * setting up log data that would be needless to compute if the logging isn't being done.
+ */
+bool LOG_SHOULD_TRACE() {
```

Comment:
i can do `LOG_SHOULD_DEBUG` etc for the levels -- no problem -- just lmk if that would be 'too much' or 'just right' :)